C++/Navio/Common/gpio.cpp

```diff
@@ -67,15 +72,13 @@ bool Pin::init()
     }
 
     uint32_t address;
-    int version = getRaspberryPiVersion();
-    if (version == 1) {
-        address = GPIO_BASE(BCM2708_PERI_BASE);
-    } else if (version == 2) {
-        address = GPIO_BASE(BCM2709_PERI_BASE);
-    } else if (version == 3) {
-        address = GPIO_BASE(BCM2835_PERI_BASE);
+
+    if (getPeriBase() == 0) {
+      return false;
     }
 
+    address = GPIO_BASE(getPeriBase());
```

Comment:
Why calling `getPeriBase` twice? it will end up calling `getRaspberryPiVersion`twice, reading through that file twice etc.

Reply:
Sure! Fixed that.

Reply:
I fixed typos, therefore I don't know how to better handle `cerr` in switch because it doesn't seem good to output two different errors or zero errors if it would find the same number as `RPI_MODEL_ERROR` in case not using a magic number you've suggested (in our case it can even be some decimal number but it doesn't look good).